### DISCUSSION TITLE
attachments

### DISCUSSION THREAD

**Discussion by vigilancetech-com** [ahyatt/ekg#120] [Q&A]
I'm not really seeing why attachments won't work.   I'm assuming they couldn't be attached to a note itself, but I don't see any reason why they couldn't be attached to a header.   A unique header ID would be generated in its property drawer and the tag :ATTACH: would be added to the header itself, and then the file would be placed in the regular org attachment directory.   When I do this with a note I just captured then I open the attachment before saving the note off it works.   If I save it off and revisit the note it says that "No attachment directory exists" however the variable `org-attach-id-dir `still contains my org data directory and the file I attached is in it, in the correct sub-folder for the header ID.

Remind me again why this doesn't work?

**Comment by vigilancetech-com** [comment]
okay, upon some further research, attachments **do work!!!**

What you have to do is set `org-attach-id-dir` to the **absolute path** of your desired attachment directory (probably, e.g. /home/kevin/org/data) and then have your cursor right on the header of in the note where you have put the attachment. 

I'm going to keep experimenting with this, but it looks very promising!

**Comment by ahyatt** [comment]
Thanks for the investigation!  This might be something worth putting in the manual, but I should probably investigate and see why things don't just work out of the box.  I no longer remember the reason.

**Reply by vigilancetech-com** [reply]
yeah, I haven't totally sorted it out yet but in some instances it seems to work and others it doesn't.

Have you seen my 2021 emacsconf presentation?

It's very similar to what you're doing, but no database other than a unix-like filesystem, and it handles the attachments automatically (in essence it's all about the attachments, in that, for example, you could just create a "note.org" file in the folder with all the other files/attachments and "tag" them all together if you wanted).

https://youtu.be/ggmfWPmse_w?si=b2Mv-erTrUcUTfOO
